examples/reduce/counter/example.py

```diff
@@ -1,7 +1,23 @@
-import aiorun
+import os
 from collections.abc import AsyncIterable
 
-from pynumaflow.reducer import Messages, Message, Datum, Metadata, AsyncReducer
+
+from pynumaflow.reducer import Messages, Message, Datum, Metadata, ReduceAsyncServer, Reducer
+
+
+class Example(Reducer):
+    async def handler(
+        self, keys: list[str], datums: AsyncIterable[Datum], md: Metadata
+    ) -> Messages:
+        interval_window = md.interval_window
+        counter = 0
+        async for _ in datums:
+            counter += 1
+        msg = (
+            f"counter:{counter} interval_window_start:{interval_window.start} "
+            f"interval_window_end:{interval_window.end}"
+        )
+        return Messages(Message(str.encode(msg), keys=keys))
```

Comment:
```suggestion
class Example(Reducer):
    def __init__(self):
        self.counter = 0
    async def handler(
        self, keys: list[str], datums: AsyncIterable[Datum], md: Metadata
    ) -> Messages:
        interval_window = md.interval_window
        counter = 0
        async for _ in datums:
            self.counter += 1
        msg = (
            f"counter:{self.counter} interval_window_start:{interval_window.start} "
            f"interval_window_end:{interval_window.end}"
        )
        return Messages(Message(str.encode(msg), keys=keys))
```

this code will not give the expected output.

Reply:
Updated to create a new instance of the reducer whenever a new key is invoked